step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/AbstractAccessor.java

```diff
@@ -131,12 +143,80 @@ private Equals byId(ObjectId id) {
 
 	@Override
 	public T save(T entity) {
+		if (versionedCollectionDriver != null) {
+			versionedCollectionDriver.save(getVersionForEntity(entity));
+		}
 		return collectionDriver.save(entity);
 	}
 
 	@Override
 	public void save(Iterable<T> entities) {
-		collectionDriver.save((Iterable<T>)entities);
+		if (versionedCollectionDriver != null) {
+			List<VersionableEntity> versionedEntities = new ArrayList<>();
+			entities.forEach(e->versionedEntities.add(getVersionForEntity(e)));
+			versionedCollectionDriver.save(versionedEntities);

```

Comment:
Could use stream().map() instead

Reply:
replaced with StreamSupport.stream of iterable to work with stream.map